app/core/src/main/java/com/fsck/k9/controller/DeleteOperationDecider.kt

```diff
@@ -0,0 +1,26 @@
+package com.fsck.k9.controller
+
+import com.fsck.k9.Account
+
+/**
+ * Decides whether deleting a message in the app moves it to the trash folder or deletes it immediately.
+ *
+ * Note: This only applies to local messages. What remote operation is performed when deleting a message is controlled
+ * by [Account.deletePolicy].
+ */
+internal class DeleteOperationDecider {
```

Comment:
👍 

Maybe `LocalDeleteOperationDecider` reflects it's purpose better or change it to an interface and provide a local implementation.